libzstd依赖问题导致编译出错

`/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/13/../../../x86_64-linux-gnu/libelf.a(elf_compress.o): in function `__libelf_compress':(.text+0x113): undefined reference to `ZSTD_createCCtx' 

出问题是因为libelf依赖libzstd，libzstd静态链接flag缺失导致编译失败。
 输入 `pkg-config -libs --static` 提示链接flag为`-lzstd -pthread`
在Makefile中增加-lzstd就能正常编译了。

--- a/common.mk
+++ b/common.mk
@@ -7,8 +7,13 @@ CFLAGS		+= -I./ -I$(ROOT)/shared/bpf/ -g
 BPF_CFLAGS	= $(CFLAGS) -Wno-unused-function			\
 		  -Wno-compare-distinct-pointer-types -Wuninitialized	\
 		  -D__TARGET_ARCH_$(SRCARCH) -DBPF_NO_PRESERVE_ACCESS_INDEX
+
+ifeq ("$(shell pkg-config --print-requires-private libelf | grep libzstd)","libzstd")
+LIBELF_ZSTD_FLAGS = -lzstd
+endif
+
 HOST_CFLAGS	= \
-		-lbpf -lelf -lz -O2 -static $(CFLAGS) -Wall		\
+		-lbpf -lelf -lz $(LIBELF_ZSTD_FLAGS) -O2 -static $(CFLAGS) -Wall		\
 		-Wno-deprecated-declarations -DVERSION=$(VERSION)	\
 		-DRELEASE=$(RELEASE)					\
 		-I$(ROOT)/shared/ -I$(ROOT)/component

